agent/src/containerutilityinterface.cpp

```diff
@@ -70,4 +78,21 @@ void ContainerUtilityInterface::removeOldContainers()
     delete containerNames;
 }
 
+void ContainerUtilityInterface::checkWorkspace()
+{
+    const std::string rootDir = m_config->getStringValue("SoftwareContainer", "shared-mounts-dir");
+    if (!isDirectory(rootDir)) {
+        log_debug() << "Container root " << rootDir << " does not exist, trying to create";
+        std::unique_ptr<CreateDir> createDirInstance = std::unique_ptr<CreateDir>(new CreateDir());
```

Comment:
and maybe initiate it like 
`std::unique_ptr<CreateDir> createDirInstance{new CreateDir()}`